### DISCUSSION TITLE
The $every operator does not work as expected when relations are blank

### DISCUSSION THREAD

**Discussion by tvacherat** [mikro-orm/mikro-orm#6187] [General]
### Describe the bug

The $every operator does not work as expected when relations are blank.
For example, when no relationship exists (OneToMany or ManyToMany), the $every operator returns the rows concerned even if they do not match the conditions defined in $every.
I deduce that $every only applies if at least one relationship exists.
Is this the desired behaviour? If so, the combination with ‘$some: {}’ may solve the problem.
This behaviour is reproduced in the given repository.

### Reproduction

https://github.com/tvacherat/Mikro-Orm-Issue

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

v6.3.13

### Node.js version

v20.10.0

### Operating system

Ubuntu 20.04.6 LTS

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
Yes, I believe this is the correct behavior, the condition is valid for all items since there are no items. 

The very same logic is there when you use the JS array methods:

```ts
[].every(item => item === 1) // true
```

**Comment by tvacherat** [comment]
From a linguistic point of view, every means that there is at least one subject, hence my question.
However, from a computing point of view, your reasoning seems to be the right one. The javascript example confirms this.
Thanks for the explanation

**Reply by B4nan** [reply]
> From a linguistic point of view, every means that there is at least one subject, hence my question.

I don't understand what linguistics has to do with programming, this is about formal logic, not language, and I still disagree with what you said here, I don't feel such implication in the word "every" at all.

I did a quick search and this actually has a formal name:

https://en.wikipedia.org/wiki/Vacuous_truth

**Reply by tvacherat** [reply]
My point is simply that it feels strange to say (and to read in the code) that everything is also nothing in this case. Not that I want to question the choices made here. Once again, the article proves that this choice is the right one.
Thank you for the knowledge
